providers/hostingde/hostingdeProvider.go

```diff
@@ -145,16 +147,25 @@ func (hp *hostingdeProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*m
 			msg = append(msg, c.String())
 		}
 
-		if len(create) == 0 && len(del) == 0 && len(mod) == 0 {
-			return nil, nil
-		}
-
-		corrections := []*models.Correction{
-			{
-				Msg: fmt.Sprintf("\n%s", strings.Join(msg, "\n")),
-				F: func() error {
-					return hp.updateRecords(dc.Name, create, del, mod)
-				},
+	corrections := []*models.Correction{
+		{
+			Msg: fmt.Sprintf("\n%s", strings.Join(msg, "\n")),
+			F: func() error {
+				for i := 0; i < 10; i++ {
+					err := hp.updateRecords(dc.Name, create, del, mod)
+					if err == nil {
+						return nil
+					}
+					// Code:10205 indicates the zone is currently blocked due to a running zone update.
+					if !strings.Contains(err.Error(), "Code:10205") {
+						return err
+					}
+
+					// Exponential back-off retry.
+					// Base of 1.8 seemed like a good trade-off, retrying for approximately 45 seconds.
+					time.Sleep(time.Duration(math.Pow(1.8, float64(i))) * 100 * time.Millisecond)
+				}
+				return fmt.Errorf("timeout: zone blocked for over 10 seconds")
```

Comment:
"10 seconds" or "10 tries"?

Reply:
Good catch! Was initially a linear retry with 10 attempts. So that's about 9 seconds to be precise :D

I'll fix that.